search_service/api/dashboard.py

```diff
@@ -0,0 +1,70 @@
+from http import HTTPStatus
+from typing import Iterable, Any
+
+from flask_restful import fields, Resource, marshal_with, reqparse  # noqa: I201
+from flasgger import swag_from
+
+from search_service.proxy import get_proxy_client
+
+
+DASHBOARD_INDEX = 'dashboard_search_index'
+
+
+# todo: Use common to produce this
+dashboard_fields = {
+    "uri": fields.String,
+    "cluster": fields.String,
+    "group_name": fields.String,
+    "group_url": fields.String,
+    "product": fields.String,
+    "name": fields.String,
+    "url": fields.String,
+    "description": fields.String,
+    "last_successful_run_timestamp": fields.Integer
+}
+
+search_dashboard_results = {
+    "total_results": fields.Integer,
+    "results": fields.Nested(dashboard_fields, default=[])
+}
+
+
+class SearchDashboardAPI(Resource):
+    """
+    Search Dashboard API
+    """
+
+    def __init__(self) -> None:
+        self.proxy = get_proxy_client()
+
+        self.parser = reqparse.RequestParser(bundle_errors=True)
+
+        self.parser.add_argument('query_term', required=True, type=str)
+        self.parser.add_argument('page_index', required=False, default=0, type=int)
+        self.parser.add_argument('index', required=False, default=DASHBOARD_INDEX, type=str)
+
+        super(SearchDashboardAPI, self).__init__()
+
+    @marshal_with(search_dashboard_results)
```

Comment:
Could we remove `marshal_with` as it's using open API (swagger)?

Reply:
it requires SearchResult to json serializable which is a bigger cleanup task than what this pr intends to be. I would prefer to do it later.

Reply:
https://github.com/lyft/amundsen/issues/379